deps/wazuh_testing/wazuh_testing/vulnerability_detector.py

```diff
@@ -95,3 +101,101 @@ def make_vuln_callback(pattern):
+        print("Failed to insert data into sqlite table\n", error)
+    finally:
+        db[1].close()
+        db[0].close()
+
+
+def insert_vulnerability(CVEID = "WVE-000", TARGET = "RHEL7", TARGET_MINOR= "", PENDING = 0, 
+    PACKAGE = "WazuhIntegrationPackage", OPERATION = "less than", OPERATION_VALUE = "2.0.0-1.el7",
+    TITLE = "", SEVERITY = "critical", PUBLISHED =  datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ"),
+    UPDATED = "", REFERENCE = "https://github.com/wazuh/wazuh-qa", TARGET_V = "REDHAT", CVSS = "10.000000", 
+    CVSS_VECTOR = "AV:N/AC:L/Au:N/C:C/I:C/A:C", RATIONALE = "Wazuh integration test vulnerability", 
+    CVSS3 = "", BUGZILLA_REFERENCE = "https://github.com/wazuh/wazuh-qa", CWE = "WVE-000 -> WVE-001", 
+    ADVISORIES = "RHSA-2010:0029"
+):
+    """
+    Insert a new vulnerability in the database of vulnerabilities, with the parameters given as arguments.
+    If used in conjunction with insert_package_in_db using the default arguments it will generate an alert.
+    """
+    try:
+        db = load_db(CVE_DB_PATH)
+        sqlite_insert_query_vulnerabilities = """INSERT INTO VULNERABILITIES
+            (CVEID, TARGET, TARGET_MINOR, PENDING, PACKAGE, OPERATION, OPERATION_VALUE)
+            VALUES 
+            (\"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\")""".format(
+                CVEID, TARGET, TARGET_MINOR, PENDING, PACKAGE, OPERATION, 
+                OPERATION_VALUE
+            )
+
+
+        sqlite_insert_query_vulnerabilities_info = """INSERT INTO VULNERABILITIES_INFO 
+            (ID, TITLE, SEVERITY, PUBLISHED, UPDATED, REFERENCE, TARGET, RATIONALE, CVSS, CVSS_VECTOR, CVSS3, \
+                BUGZILLA_REFERENCE, CWE, ADVISORIES)
+            VALUES 
+            (\"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \
+            \"{}\")""".format(
+                CVEID, TITLE, SEVERITY, PUBLISHED, UPDATED, REFERENCE, TARGET_V, RATIONALE, CVSS, CVSS_VECTOR, CVSS3, 
+                BUGZILLA_REFERENCE, CWE, ADVISORIES
+            )
+        db[1].execute(sqlite_insert_query_vulnerabilities)
+        db[1].execute(sqlite_insert_query_vulnerabilities_info)
+        db[0].commit()
+    except sqlite3.Error as error:
+        print("Failed to insert data into sqlite table\n", error)
+    finally:
+        db[1].close()
+        db[0].close()
+
+
+def update_package(version, package, agent = "000"):
+    """
+    Update version of installed package in databas.
+    Used to simulate upgrades and downgrades of the package given as argument.
+    """
+    try:
+        path =  os.path.join(PACKAGES_DB_PATH,"{}.db".format(agent))
+        db = load_db(path)
+        sqlite_update_query = "update sys_programs set version=\"{}\" where name=\"{}\";".format(version, package)
+        db[1].execute(sqlite_update_query)
+        db[0].commit()
+    except sqlite3.Error as error:
+        print("Failed to insert data into sqlite table\n", error)
```

Comment:
This is pretty similar to the last comment, refactor this code in a new function and don't duplicate errors.

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/700/commits/4d232de50ad20cca3108cd2423524a614fa93475